src/engine/Join.cpp

```diff
@@ -144,6 +144,21 @@ void Join::computeResult(ResultTable* result) {
                     _leftJoinCol, rightRes->_idTable, _rightJoinCol,
                     &result->_idTable);
 
+  // If only one of the two operands has a local vocab, pass it on.
+  bool leftLocalVocabEmpty = leftRes->_localVocab->empty();
+  bool rightLocalVocabEmpty = rightRes->_localVocab->empty();
+  if (!leftLocalVocabEmpty || !rightLocalVocabEmpty) {
+    if (!leftLocalVocabEmpty && rightLocalVocabEmpty) {
+      result->_localVocab = std::move(leftRes->_localVocab);
+    } else if (leftLocalVocabEmpty && !rightLocalVocabEmpty) {
+      result->_localVocab = std::move(rightRes->_localVocab);
```

Comment:
I think those are `shared_ptrs` that should be copied, not moved. Otherwise, the cache might be invalidated for the `leftRes` and `rightRes`.